### PR TITLE
rust: make clippy happy

### PR DESCRIPTION
**Pull request by ignatenkobrain**
Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>

### CODE REVIEW

File: rust/src/main.rs [hunk 1]
```diff
@@ -42,22 +42,22 @@ impl EvTag {
         let object = odb.read(object.id())?;
         let contentbuf = object.data();
         let header = format!("{} {}", object.kind().str(), contentbuf.len());
-        hash.write(header.as_bytes()).unwrap();
+        hash.write_all(header.as_bytes()).unwrap();
```
Comment (cgwalters) [inline rust/src/main.rs]: Oh, that's a useful catch indeed, though I'd be surprised if any of the hashers actually did partial writes.

File: rust/src/main.rs [hunk 2]
```diff
@@ -42,22 +42,22 @@ impl EvTag {
...
+        hash.write_all(contentbuf).unwrap();
         Ok(())
     }
 
     fn checksum_submodule(&self, hash: &mut Hasher, submodule: &Submodule) -> Result<(), Error> {
         let subrepo = submodule.open()?;
         let sub_head = submodule.workdir_id().expect("Failed to find workdir id");
         let commit = subrepo.find_commit(sub_head)?;
-        self.checksum_commit_contents(&subrepo, commit, hash)?;
+        self.checksum_commit_contents(&subrepo, &commit, hash)?;
```
Comment (cgwalters) [inline rust/src/main.rs]: Hum interesting.  Is this like "unnecessary move" or something?

Comment (ignatenkobrain) [reply rust/src/main.rs]: I think it was https://rust-lang-nursery.github.io/rust-clippy/master/index.html#needless_pass_by_value

Comment (cgwalters) [reply rust/src/main.rs]: Ah yep.  That's a cool lint.

**Pull request merged**